slurp/src/main/webapp/scripts/utils.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Parse URL <pre><code>NAME.html?key1=value1&key2=value2...</code></pre> into {key1: value1, ...}.
+ * @param {string} url The URL to be parsed. 
+ * @returns {Object.<string, string>} Dictionary of the parsed URL parameters. 
+ */
+function parseUrl(url) {
+  var paramsUrl = url.substring(url.indexOf("?") + 1);
+  var paramPairs = paramsUrl.split("&");
+  
+  var params = {};
+  for (var pair of paramPairs) {
+    var splitIndex = pair.indexOf("=");
+    
+    var key = pair.substring(0, splitIndex);
+    var value = pair.substring(splitIndex+1);
+
+    params[key] = value;
+  }
+
+  return params;
+}
+
+/**
+ * Make a random 32-digit UUID. 
+ * @returns {string} 32-digit UUID as string.
+ */
+function getRandomUuid() {
```

Comment:
Just curious, what will we need this for? If you do not specify a document Id, firebase should make one automatically. See [here](https://firebase.google.com/docs/firestore/manage-data/add-data#add_a_document).

Reply:
I've been using it when I auto-generate divs.

Reply:
I'm using it so I can auto-generate IDs for divs.